asmcomp/un_anf.ml

```diff
@@ -156,7 +156,7 @@ let make_var_info (clam : Clambda.ulambda) : var_info =
       List.iter loop args
     | Ucatch (static_exn, vars, body, handler) ->
       ignore_int static_exn;
-      ignore_var_with_provenance_list vars;
+      ignore_var_with_provenance_list (List.map fst vars);
```

Comment:
That function does nothing but asserts the type. you should change the definition instead:
```
let ignore_var_with_provenance_list (_ : (VP.t  * value_kind) list) = ()
```

Reply:
The function was used also for functions parameters.  I've fixed that by keeping  value kinds on those parameters in Clambda as well :-) (and also for the function return).